Tests/Storage/TestSQLiteEventStore.swift

```diff
@@ -60,52 +52,54 @@ class TestSQLiteEventStore: XCTestCase {
         payload.addValueToPayload("Welcome to foobar!", forKey: "page")
         payload.addValueToPayload("MEEEE", forKey: "refr")
 
-        for _ in 0..<250 {
-            _ = eventStore.insertEvent(payload)
+        let dispatchQueue = DispatchQueue(label: "Save events", attributes: .concurrent)
+        let expectations = [
+            XCTestExpectation(),
+            XCTestExpectation(),
+            XCTestExpectation(),
+            XCTestExpectation(),
+            XCTestExpectation()
+        ]
+        for i in 0..<5 {
+            dispatchQueue.async {
+                for _ in 0..<250 {
+                    eventStore.addEvent(payload)
+                }
+                expectations[i].fulfill()
+            }
         }
-
-        XCTAssertEqual(eventStore.count(), 250)
-        XCTAssertEqual(eventStore.getAllEventsLimited(600)?.count, 250)
-        XCTAssertEqual(eventStore.getAllEventsLimited(150)?.count, 150)
-        XCTAssertEqual(eventStore.getAllEvents()?.count, 250)
-
+        wait(for: expectations, timeout: 10)
+        
+        XCTAssertEqual(eventStore.count(), 1250)
+        XCTAssertEqual(eventStore.emittableEvents(withQueryLimit: 600).count, 250)
+        XCTAssertEqual(eventStore.emittableEvents(withQueryLimit: 150).count, 150)
+        
         _ = eventStore.removeAllEvents()
         XCTAssertEqual(eventStore.count(), 0)
     }
 
     func testSQLiteEventStoreCreateSQLiteFile() {
-        _ = SQLiteEventStore(namespace: "aNamespace")
+        let eventStore = createEventStore("aNamespace")
+        _ = eventStore.count()
```

Comment:
what's this count for?

Reply:
This is actually no longer necessary – I added that because at one time I wanted to initialize the database lazily (only create it on the first database call), but I have later changed my mind and create it during initialization. I have removed this from the tests, thanks for pointing out!